packages/dids/src/did-dht.ts

```diff
@@ -0,0 +1,225 @@
+      ...relationshipsMap,
+      ...services && {service: services}
+    };
+
+    if (publish) {
+      await this.publish(keySet, document);
+    }
+    return {
+      did      : document.id,
+      document : document,
+      keySet   : keySet
+    };
+  }
+
+  public static async publish(keySet: DidDhtKeySet, didDocument: DidDocument): Promise<DidResolutionResult> {
+    const dht = new DidDht();
+    const publicCryptoKey = await Jose.jwkToCryptoKey({key: keySet.identityKey.publicKeyJwk});
+    const privateCryptoKey = await Jose.jwkToCryptoKey({key: keySet.identityKey.privateKeyJwk});
+
+    const request = await dht.createPutDidRequest({
+      publicKey  : publicCryptoKey,
+      privateKey : privateCryptoKey
+    }, didDocument);
+
+    const hash = await dht.put(request);
+    if (hash) {
+      return {
+        didDocumentMetadata   : undefined,
+        didDocument,
+        didResolutionMetadata : {
+          contentType: 'application/json'
+        }
+      };
+    } else {
+      throw new Error('Failed to publish to DHT');
+    }
+  }
+
+  public static async resolve(did: string): Promise<DidDocument> {
+    const dht = new DidDht();
+    const hash = did.replace('did:dht:', '');
+    const decoded = z32.decode(hash);
+    const identifier = this.hash(decoded);
+    const retrievedValue = await dht.get(identifier.toString('hex'));
+    return await dht.parseGetDidResponse(did, retrievedValue).finally(() => dht.destroy());
+  }
+
+  private static hash(input: crypto.BinaryLike) {
+    return crypto.createHash('sha1').update(input).digest();
+  }
+
+  public static async getDidIdentifier(options: {
+        key: PublicKeyJwk
+    }): Promise<string> {
+    const {key} = options;
+
+    const cryptoKey = await Jose.jwkToCryptoKey({key});
+    const identifier = z32.encode(cryptoKey.material);
+    return 'did:dht:' + identifier;
+  }
+
+  public static async getDidIdentifierFragment(options: {
+    key: PublicKeyJwk
+  }): Promise<string> {
+    const {key} = options;
+    const cryptoKey = await Jose.jwkToCryptoKey({key});
+    return z32.encode(cryptoKey.material);
+  }
+
+  public static async generateJwkKeyPair(options: {
+        keyAlgorithm: typeof SupportedCryptoKeyTypes[number],
+        keyId?: string
+    }): Promise<JwkKeyPair> {
+    const {keyAlgorithm, keyId} = options;
+
+    let cryptoKeyPair: Web5Crypto.CryptoKeyPair;
+
+    switch (keyAlgorithm) {
+      case 'Ed25519': {
+        cryptoKeyPair = await new EdDsaAlgorithm().generateKey({
+          algorithm   : {name: 'EdDSA', namedCurve: 'Ed25519'},
+          extractable : true,
+          keyUsages   : ['sign', 'verify']
+        });
+        break;
+      }
+
+      case 'secp256k1': {
+        cryptoKeyPair = await new EcdsaAlgorithm().generateKey({
+          algorithm   : {name: 'ECDSA', namedCurve: 'secp256k1'},
+          extractable : true,
+          keyUsages   : ['sign', 'verify']
+        });
+        break;
+      }
```

Comment:
nothing immediate just wanted to call out that the same code exists [here](https://github.com/TBD54566975/web5-js/blob/main/packages/dids/src/did-key.ts#L644-L666) for `did:key` and [here](https://github.com/TBD54566975/web5-js/blob/f3c7184bab45972821c89d6275eda8a6c295884f/packages/dids/src/did-ion.ts#L260-L282) for `did:ion`

@frankhinek thoughts on eventually adding a `Web5SubleCrypto` api surface that matches [W3C WebCrypto's SubtleCrypto](https://www.w3.org/TR/WebCryptoAPI/#subtlecrypto-interface)?